### PR TITLE
Soften wording on IR compiler migration guide re nullable Booleans

### PR DESCRIPTION
**Pull request by SebastianAigner**
As per #2816

### CODE REVIEW

File: docs/topics/js/js-ir-migration.md [hunk 1]
```diff
@@ -214,4 +170,52 @@ kotlin {
...
+
+```kotlin
+external interface ComponentProps: Props {
+   var isInitialized: Boolean
+   var visible: Boolean
+}
+```
+
+```kotlin
+val props = js("{}") as ComponentProps
```
Comment (turansky) [inline docs/topics/js/js-ir-migration.md:192]: `val props: ComponentProps = js("({})")`

without redundant cast

File: docs/topics/js/js-ir-migration.md [hunk 2]
```diff
@@ -214,4 +170,52 @@ kotlin {
...
+props.isInitialized = true
+// visible is not initialized - OK in JS – means it's false
+```
+
+If you try to use such a property in a function overridden in Kotlin (for example, a React `button`), you'll get a `ClassCastException`:
+
+```kotlin
+button {
+   attrs {
```
Comment (turansky) [inline docs/topics/js/js-ir-migration.md:201]: No more `attrs`?

File: docs/topics/js/js-ir-migration.md [hunk 3]
```diff
@@ -214,4 +170,52 @@ kotlin {
...
+       autoFocus = props.visible // ClassCastException here
```
Comment (turansky) [inline docs/topics/js/js-ir-migration.md:202]: `ClassCastException` isn't expected here:
* `autoFocus` property in nullable
* `null` works as `false`

**Review by turansky — CHANGES_REQUESTED** [review]

**Comment by SebastianAigner** [comment]
Further softened the wording. As per discussion with @turansky, turns out that the error surface is smaller than initially anticipated – in all cases (besides calling `toString`) `null` and `undefined` booleans behave correctly in Kotlin/JS, so I've made the warning specific to calling toString.

@turansky @bashor PTAL if time allows :)

File: docs/topics/js/js-ir-migration.md [hunk 4]
```diff
@@ -214,4 +170,39 @@ kotlin {
...
+
+**Issue**: when you call `toString` on a `Boolean` from an external interface, you're getting an error like `Uncaught TypeError: Cannot read properties of undefined (reading 'toString')`. JavaScript treats the `null` or `undefined` values of a boolean variable as `false`. If you rely on calling `toString` on a `Boolean` that may be `null` or `undefined` (for example when your code is called from JavaScript code you have no control over), be aware of this:
+
+```kotlin
+external interface SomeExternal {
+    var visible: Boolean
+}
+
+fun main() {
+    val empty: SomeExternal = js("{}") as SomeExternal
```
Comment (turansky) [inline docs/topics/js/js-ir-migration.md:189]: `as SomeExternal` is redundant

**Review by turansky — CHANGES_REQUESTED** [review]

File: docs/topics/js/js-ir-migration.md (hunk 4 above)
Comment (SebastianAigner) [reply docs/topics/js/js-ir-migration.md]: ...finger must've slipped when copying this snippet from my sample project, d'oh 😱

**Review by turansky — APPROVED** [review]

File: docs/topics/js/js-ir-migration.md [hunk 5]
```diff
@@ -214,4 +170,39 @@ kotlin {
...
 }
+```
+
+## Additional troubleshooting tips when working with the Kotlin/JS IR compiler
+
+These hints may help you when troubleshooting problems in your projects using the Kotlin/JS IR compiler.
+
+### Make boolean properties nullable in external interfaces
+
+**Issue**: when you call `toString` on a `Boolean` from an external interface, you're getting an error like `Uncaught TypeError: Cannot read properties of undefined (reading 'toString')`. JavaScript treats the `null` or `undefined` values of a boolean variable as `false`. If you rely on calling `toString` on a `Boolean` that may be `null` or `undefined` (for example when your code is called from JavaScript code you have no control over), be aware of this:
```
Comment (bashor) [inline docs/topics/js/js-ir-migration.md:181]: it, also, could be fail in casts to `Boolean` and `Any` (note not-null!) either explicit and implicit. Implicit ones could be generated by compiler, e.g. when refine generic type in override  (bridges), like:
https://pl.kotl.in/-NnNLbtOQ
```kotlin
external interface Foo {
    val bar: Boolean
}

interface A<T> {
    fun foo(a: T) = a
}

class B : A<Boolean> {
    override fun foo(a: Boolean) = a
}


fun main() {
    try {
        val o: Foo = js("{}")
	println("1")
        val bool = o.bar// as Boolean
        println("2")
        val a = B() as A<Boolean>
        println("3")
        println(a.foo(bool))
        println("done")
    } catch (e: Throwable) {
        println("fail")
    }
}
```

Comment (turansky) [reply docs/topics/js/js-ir-migration.md:181]: `Boolean` as type parameter doesn't look like typical case :)

**Comment by turansky** [comment]
Merge?

**Pull request merged**